src/main/java/kmql/QueryCompleter.java

```diff
@@ -0,0 +1,61 @@
+package kmql;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.jline.reader.Candidate;
+import org.jline.reader.Completer;
+import org.jline.reader.LineReader;
+import org.jline.reader.ParsedLine;
+import org.jline.reader.impl.completer.StringsCompleter;
+
+/**
+ * A {@link Completer} implementation for interactive kmql session.
+ */
+public class QueryCompleter implements Completer {
+    private final StringsCompleter delegate;
+
+    public static QueryCompleter from(Engine engine) {
+        return new QueryCompleter(engine, TableRegistry.DEFAULT);
+    }
+
+    QueryCompleter(Engine engine, TableRegistry tableRegistry) {
+        Set<String> candidates = new HashSet<>();
+
+        for (Entry<String, Table> entry : tableRegistry) {
+            candidates.add(entry.getKey().toLowerCase());
+            candidates.addAll(engine.showColumns(entry.getValue())
```

Comment:
hm.. this could be bit confusing because for example, a word "partition" could be suggseted when attempting to complete at `select * from brokers where [TAB]`. while the brokers table has no partition column, completion suggests it... 
Extremely it's better to parse sql query and suggest words that are contextually valid only, but better than have nothing so should be okay now 👍

Reply:
Yeah agree. It'd better to support contextual suggestion.
But I guess there's a difficulty in doing that for SQL because of its syntax.

completer should suggest columns when writing `select topic, partition,` but available columns could not be determined until writing `from` ...